syntax = "proto3";

package capsule;

import "capsule/common.proto";

message CrawlRequest {
    repeated string urls = 1;
}

message CrawlResponse {
    string url = 1;
    Content content = 2;
    bool failed = 3;
}

service Worker {
    rpc Crawl(CrawlRequest) returns (stream CrawlResponse);
}
